ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -159,6 +159,205 @@ objects:
+          done
+
+          echo "Collect etcd names"
+          ETCD_CONN_STRING_TEMP=""
+          for master in ${MASTERS[@]}
+          do
+            ETCD_NAME="etcd-member-$(bastion_ssh core@${master} hostname -f)"
+            ETCD_URI=$(bastion_ssh core@${master} 'source /run/etcd/environment && echo "https://${ETCD_DNS_NAME}:2380"')
+            ETCD_CONN_STRING_TEMP="${ETCD_CONN_STRING_TEMP}${ETCD_NAME}=${ETCD_URI},"
+          done
+          ETCD_CONN_STRING=${ETCD_CONN_STRING_TEMP::-1}
+
+          echo "Stop static pods"
+          for master in ${MASTERS[@]}
+          do
+            bastion_ssh core@${master} "sudo -i mkdir /etc/kubernetes/stopped-pods && sudo -i mv /etc/kubernetes/manifests/* /etc/kubernetes/stopped-pods"
+          done
+
+          echo "Restore etcd from snapshot"
+          cat > /tmp/etcd_restore.sh <<'EOF'
+            #!/bin/bash
+            set -ex
+
+            ETCD_VER=v3.3.10
+            DOWNLOAD_URL=https://storage.googleapis.com/etcd
+            ASSET_DIR=/root/.local
+
+            mkdir -p ${ASSET_DIR}/bin ${ASSET_DIR}/tmp ${ASSET_DIR}/shared
+
+            echo "Downloading etcdctl binary.."
+            curl -s -L ${DOWNLOAD_URL}/${ETCD_VER}/etcd-${ETCD_VER}-linux-amd64.tar.gz -o $ASSET_DIR/tmp/etcd-${ETCD_VER}-linux-amd64.tar.gz \
+              && tar -xzf $ASSET_DIR/tmp/etcd-${ETCD_VER}-linux-amd64.tar.gz -C $ASSET_DIR/shared --strip-components=1 \
+              && mv $ASSET_DIR/shared/etcdctl $ASSET_DIR/bin/ \
+              && rm $ASSET_DIR/shared/etcd \
+              && ETCDCTL_API=3 $ASSET_DIR/bin/etcdctl version
+
+            ETCD_NAME="etcd-member-$(hostname -f)"
+
+            rm -rf /var/lib/etcd/
+
+            source /run/etcd/environment
+            ETCDCTL_API=3 etcdctl \
+              --cert /etc/ssl/etcd/system:etcd-peer:${ETCD_DNS_NAME}.crt \
+              --key /etc/ssl/etcd/system:etcd-peer:${ETCD_DNS_NAME}.key \
+              --cacert /etc/ssl/etcd/ca.crt \
+              snapshot restore /tmp/snapshot.db \
+              --name ${ETCD_NAME} \
+              --initial-cluster "$@" \
+              --initial-cluster-token etcd-cluster-1 \
+              --skip-hash-check=true \
+              --initial-advertise-peer-urls https://${ETCD_IPV4_ADDRESS}:2380 \
+              --data-dir /var/lib/etcd/
+        EOF
+          chmod +x /tmp/etcd_restore.sh
+          for master in ${MASTERS[@]}
+          do
+            scp -o StrictHostKeyChecking=no -o ProxyCommand="ssh -A -o StrictHostKeyChecking=no -o ServerAliveInterval=30 -W %h:%p core@${BASTION_HOST}" /tmp/etcd_restore.sh core@${master}:/tmp
+            bastion_ssh core@${master} "sudo -i /tmp/etcd_restore.sh ${ETCD_CONN_STRING}"
+          done
+
+          for master in ${MASTERS[@]}
+          do
+            while true
+            do
+              bastion_ssh core@${master} "sudo -i mv /etc/kubernetes/manifests-backup/* /etc/kubernetes/manifests" && break; sleep 10
```

Comment:
There's lots of SSHing into the bastion, exiting, and then immediately SSHing back in and doing something else.  Can we SSH up once with some big script to do all of this?  I'm fine landing this PR without that change, but I think it might make things easier to follow.

Reply:
Ideally - yes, it would be great to merge some scripts. In this particular case we have to restore etcd from the snapshot on all three masters before starting apiservers there.